### PR TITLE
Add support for buttons in buffers created by dictionary.el

### PR DESCRIPTION
**Pull request by mattbeshara**
This lets you use Link Hints for opening buffers with searches for definitions of related words.

### CODE REVIEW

File: link-hint.el [hunk 1]
```diff
@@ -59,6 +59,7 @@
     link-hint-gnus-w3m-url
     link-hint-gnus-w3m-image-url
     link-hint-deadgrep
+    link-hint-dictionary-button
```
Comment (noctuid) [inline link-hint.el:62]: Could you also add a bullet in the readme for the new type?

File: link-hint.el [hunk 2]
```diff
@@ -1076,6 +1078,23 @@ Only search the range between just after the point and BOUND."
...
   :copy #'link-hint--copy-xref-item)
 
+;; ** Dictionary button
+(defun link-hint--next-dictionary-button (&optional end-bound start-bound)
+  (let ((pos (next-single-char-property-change (or start-bound (point))
+                                               'button
+                                               nil
+                                               end-bound)))
+    (when (< pos end-bound)
+      (if-let ((button (button-at pos)))
```
Comment (noctuid) [inline link-hint.el:1088]: Link-hint currently supports Emacs 24.4, so don't use if-let (requires >=Emacs 25.1).

File: link-hint.el [hunk 3]
```diff
@@ -1076,6 +1078,23 @@ Only search the range between just after the point and BOUND."
...
+          (overlay-start button)
+        (link-hint--next-dictionary-button end-bound pos)))))
+
+(link-hint-define-type 'dictionary-button
+  :next #'link-hint--next-dictionary-button
+  :vars '(dictionary-mode)
```
Comment (noctuid) [inline link-hint.el:1094]: Also add dictionary-mode to the generic button type's :not-vars.

**Review by noctuid — COMMENTED** [review]
I think the code for "woman button" will already work as-is for dictionary.el. If it does, maybe combine the two by just renaming `link-hint-woman-button` to something like `link-hint-special-button` and using `:vars '(woman-mode dictionary-mode)`.

File: link-hint.el (hunk 1 above)
Comment (mattbeshara) [reply link-hint.el]: I've mentioned Dictionary mode in the bullet for 'buttons', which is where Woman and Man modes are currently listed.

File: link-hint.el (hunk 2 above)
Comment (mattbeshara) [reply link-hint.el]: I was able to delete this function entirely.

File: link-hint.el (hunk 3 above)
Comment (mattbeshara) [reply link-hint.el]: I've added both dictionary and Man mode, as it turned out that Man mode also used the exact same overlay button code as Woman mode.  I've consolidated Woman, Man, and dictionary types into a single 'overlay button' type in the commits I've just pushed.  Let me know if you think that's a good idea or whether I should separate them back out.

**Comment by noctuid** [comment]
Great, thanks!

**Pull request merged**